src/coreclr/vm/ilstubcache.cpp

```diff
@@ -553,6 +561,12 @@ MethodDesc* ILStubCache::GetStubMethodDesc(
             SigTypeContext::InitTypeContext(pTargetMD, &typeContext);
         }
 
+        if (pTypeContext != NULL)
```

Comment:
We have just computed a type context from pTargetMD and we are going to throw it away. It would be better to do:
```
SigTypeContext typeContext;
if (pTypeContext == NULL)
{
    if (pTargetMD != NULL)
    {
        SigTypeContext::InitTypeContext(pTargetMD, &typeContext);
    }
    pTypeContext = &typeContext;
}
```
and use pTypeContext in the rest of the method.